source/Lexer.cpp

```diff
@@ -220,6 +223,12 @@ Token Lexer::lex(LexerMode mode, KeywordVersion keywordVersion) {
 
     onNewLine = false;
     info->rawText = lexeme();
+    if (diagnostics.count() >= MAX_LEXER_ERRORS) {
+        // Stop any further lexing by claiming to at the end of the buffeer;
+        addError(DiagCode::TooManyErrors, currentOffset());
+        // Ensure we don't look at the source any more
+        sourceBuffer = nullptr;
```

Comment:
Rather than clearing out the sourceBuffer, instead I think we should skip from the current location to the end of the buffer and turn all that skipped into a Trivia object on the EoF token we create. That way we 1) preserve the text if the caller wants to look at it for some reason and 2) we don't need an additional branch above on sourceBuffer.